src/in_progress.rs

```diff
@@ -1110,3 +1110,132 @@ pub fn is_hardware_waiting_repair(
+pub fn get_region_id(pool: &Pool<ConnectionManager>, region_name: &str) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get region Id from region name
+    let stmt = "SELECT region_id FROM regions WHERE region_name = $1";
+    let stmt_query = conn.query(stmt, &[&region_name])?;
+
+    let mut region_id: u32 = 0;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return region_id
+        let id: i32 = res.get(0);
+        region_id = id as u32;
+        debug!("Region id {} for the region {}", region_id, region_name);
+        Ok(Some(region_id))
+    } else {
+        // does not exist
+        debug!("No region with name {} in database", region_name);
+        Ok(None)
+    }
+    
+}
+
+/// Get storage id based on the storage type.
+pub fn get_storage_id(pool: &Pool<ConnectionManager>, storage_type: &str) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get storage Id from storage type
+    let stmt = "SELECT storage_id FROM storage_types WHERE storage_type= $1 ";
+    let stmt_query = conn.query(&stmt, &[&storage_type])?;
+
+    let mut storage_id: u32 = 0;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return storage_id
+        let id: i32 = res.get(0);
+        storage_id = id as u32;
+        debug!(
+            "Storage id {} for the storage_type {}",
+            storage_id, storage_type
+        );
+        Ok(Some(storage_id))
+    } else {
+        // does not exist
+        debug!("No storage with type {} in database", storage_type);
+        Ok(None)
+    }
+}
+
+/// Get storage id based on the storage type.
+pub fn get_storage_detail_id(
+    pool: &Pool<ConnectionManager>,
+    storage_id: u32,
+    region_id: u32,
+    host_name: &str,
+) -> BynarResult<Option<u32>> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get storage detail Id
+    let stmt = "SELECT detail_id FROM storage_details WHERE storage_id = $1
+            AND region_id = $2 AND hostname = $3 ";
+    let stmt_query = conn.query(&stmt, &[&storage_id,&region_id, &host_name])?;
+
+    let mut storage_detail_id: u32 = 0;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return storage_id
+        let id: i32 = res.get(0);
+        storage_detail_id = id as u32;
+        debug!(
+            "Storage details id {} for the host_name {} , region {} , storage_id {} ",
+            storage_detail_id, host_name, region_id, storage_id
+        );
+        Ok(Some(storage_detail_id))
+    } else {
+        // does not exist
+        debug!(
+            "No storage detail id with host_name {} , region {} , storage_id {} 
+        in database",
+            host_name, region_id, storage_id,
+        );
+        Ok(None)
+    }
+}
+
+/// Get a list of ticket IDs (JIRA/other ids) that belong to all servers.
+/// that are in pending state  and outstanding tickets
+pub fn get_all_pendout_tickets(
```

Comment:
did you mean: `get_all_pending_tickets` here?

Reply:
yes